### ISSUE TITLE
Create a single TestReport for the solution

### ISSUE THREAD

**Issue by palexcast** [kurtmkurtm/LiquidTestReports#13]
Would be nice to be able to create a single TestReport for the whole solution. 
Not all solutions contain 1 test project.

BTW, really good job :)!

**Comment by kurtmkurtm** [comment]
Hi, thanks!!

I will take a look to see if this is possible with the VS test integration.

I have also been slowly working on Cli tool to include in this project. This will be able convert a single or multiple trx files into a report.

**Comment by NightOwl888** [comment]
Hi there. We are also interested in combining multiple `.trx` reports into a single report, as our tests are slow and we want to run each project on a separate job. What is the status of this feature?

**Comment by kurtmkurtm** [comment]
Hi,

I have been a bit short on time to work on this recently, but I'm expecting to have this feature published by early December.

The current status is:

- [x] Refactor the drops models and mapping in the core project
- [x] Create the .NET CLI tool
- [x] Load trx files using `vstst.xsd` schema
- [ ] Add mapping from trx models to drops models

**Comment by NightOwl888** [comment]
One feature I would like to see for merging TRX reports into a single Markdown/HTML report is the ability to specify a title per TRX file and when the title is specified the report is grouped by TRX with the title as the grouping.

For example, in Azure DevOps, the default behavior when using the [Publish Test Results task](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/test/publish-test-results?view=azure-devops&tabs=trx%2Cyaml) you upload TRX files by glob pattern is to lump the entire set of TRX reports into a single summary section. However, it also supports `testRunTitle` which allows you to specify a title for the TRX file (unfortunately, you have to upload the files one at a time to use this feature). Without having a title for the TRX file, all of the context information about:

1. Assembly
2. Operating System
3. Target Framework
4. Platform (x86 or x64)

...or other such information that might be useful to repeat the failure is simply lost.

However, when a title is specified, it is used as a subheading to group the tests:

![image](https://user-images.githubusercontent.com/1538288/100013916-35abce00-2e08-11eb-9004-eafc793534dd.png)

IMO, the number of tests that failed, passed, and were skipped should be baked into the group header (in Azure DevOps we had to parse the information out of the report and add it to the title). But the title itself should be a string that can be specified for each TRX file (ideally, through the command line, but somehow adding the information to the TRX report or using a convention-based file name that is picked up automatically would also work).

Having it hide all but the tests that fail by default with the ability to expand each group and also to remove the filters to see all of the tests are also handy features. [See this pipeline to play around with the filters and expanding groups](https://dev.azure.com/LuceneNET-Temp/Lucene.NET/_build/results?buildId=1237&view=ms.vss-test-web.build-test-results-tab).

Of course, having it summarize all of the failures is also important. But merging the reports into one lump without being able to see what conditions we need to reproduce the failure is a deal-breaker for us.

**Comment by kurtmkurtm** [comment]
Hi All, 
After having a busy time the last few months, I've finally been able to get back to this.

I've published a [pre-release NuGet package](https://www.nuget.org/packages/LiquidTestReports.Cli/) of the global tool from the feature branch. 

@NightOwl888 thanks for the suggestions, I've included the ability to merge results under a custom group title, and added an option to add a suffix to test names. Because filtering isn't really possible with markdown, I have separated the results lists by outcome to avoid needing to scroll through passed tests to find failures.

I've added some sample scenarios and out outputs into the readme here:
https://github.com/kurtmkurtm/LiquidTestReports/tree/docs/readme#liquidtestreportscli-beta
